src/Compiler/CompilerOptions.cpp

```diff
@@ -92,6 +92,7 @@ OptReport optReport;                                   // onnx-mlir only
 bool useOldBufferization;                              // onnx-mlir only
 bool enableTiming;                                     // onnx-mlir only
 bool enableBoundCheck;                                 // onnx-mlir only
+int compilationNumThreads;                             // onnx-mlir only
```

Comment:
Can we apply this option to `onnx-mlir-opt` as well so that we can check performance by calling `onnx-mlir-opt --constprop-onnx -j8` for a single operation? It's ok if it needs another PR.

Reply:
@tungld I was thinking along the same lines, also for example to test that we get the same result for a test that uses parallelism.

@imaihal I would then add a test that tries the operation that you parallelized and try it with `-j 4` for example. You may have to copy a large constant to do that... can lift it from a model.

Reply:
@tungld @AlexandreEichenberger Thanks for the comments.

I applied this option to `onnx-mlir-opt`, but parallelization of constprop is not included in this PR (It is included in PR #3042). So I want to add the tests in PR #3042.

Reply:
Absolutely, this PR set the options, and it will be tested once there is a new opt that can use parallelism.

Reply:
@imaihal I didn't see any change to `onnx-mlir-opt.cpp`. I am curious on how this option is effective in`onnx-mlir-opt` command line?

Reply:
@tungld New thread pool is created in `MlirOptMain.cpp` of MLIR. This is due to handle`-split-input-file` option  (https://github.com/llvm/llvm-project/blob/main/mlir/lib/Tools/mlir-opt/MlirOptMain.cpp#L609-L613). So, it seems to be difficult to set our own threadpool in `onnx-mlir-opt.cpp`. 

In lit tests, how about using`--mlir-disable-threading` for single thread and using mlir default(use all cpus) for multi-thread? Or, using pass local option might be possible to specify our own thread pool.

Reply:
@tungld Sorry, it is possible just by modifying onnx-mlir-opt.cpp. I updated.